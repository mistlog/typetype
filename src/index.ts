export * from "./parser";
export * from "./generator";
export * from "./core";