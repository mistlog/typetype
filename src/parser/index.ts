export * from "./common";
export * from "./basic";
export * from "./function";
export * from "./expression";
export * from "./statement";
export * from "./declaration";
export * from "./start";