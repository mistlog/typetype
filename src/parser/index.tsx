export * from "./common";
export * from "./basic";
export * from "./expression";
export * from "./statement";
export * from "./declaration";
export * from "./start";