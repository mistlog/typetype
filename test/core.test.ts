import { transform } from "../src";

test("transform", () => {
    const result = transform(`
        type temp = string
        type function func = (T) => {a: T}
    `);
    expect(result).toMatchSnapshot();
})

test("transform: backtrace", () => {
    try {
        transform(`
            type temp = string
            type function func = (T) = {a: T}
        `);
    } catch (error) {
        expect(JSON.stringify(error, null, 4)).toMatchSnapshot();
    }

    try {
        transform(`
            type temp = string
            type function func = (T) = {a: T}
        `, { debug: true });
    } catch (error) {
        expect(JSON.stringify(error, null, 4)).toMatchSnapshot();
    }
})