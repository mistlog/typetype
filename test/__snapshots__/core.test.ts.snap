// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform 1`] = `
Object {
  "code": "type temp = string;
type func<T> = {
  a: T;
};",
}
`;

exports[`transform: backtrace 1`] = `
"{
    \\"backtrace\\": \\"No backtrace found. Make sure you use \`pegjs --trace\` to build your parser javascript.\\\\nOr, the failure might occur in the start node.\\"
}"
`;

exports[`transform: backtrace 2`] = `
"{
    \\"backtrace\\": \\"x 3:38-3:38 MultiLineComment\\\\n|             type function func = (T) = {a: T}\\\\n|                                      ^\\\\no 3:37-3:38 _\\\\n|             type function func = (T) = {a: T}\\\\n|                                     ^\\\\nx 3:34-3:34 TypeArrowFunctionExpression\\\\n|             type function func = (T) = {a: T}\\\\n|                                  ^\\\\nx 3:27-3:27 TypeFunctionDeclarator\\\\n|             type function func = (T) = {a: T}\\\\n|                           ^\\\\nx 3:13-3:13 TypeFunctionDeclaration\\\\n|             type function func = (T) = {a: T}\\\\n|             ^\\\\no 1:1-3:13 TypeFile\\\\n  \\\\n  \\\\n              type temp = string\\\\n              type function func = (T) = {a: T}\\\\n  ^^^^^^^^^^^^\\"
}"
`;
